File: task_2/code/fish.cpp
//
// Created by Efim Verzakov on 16.10.2021.
//

#include "fish.h"

void Fish::In(std::ifstream &ifst) {
    ifst>>this->name>>this->weight;
    int type;
    ifst >> type;
    if (type == 1) {
        p = Fish::RIVER;
    } else if (type == 2) {
        p = Fish::SEA;
    } else {
        p = Fish::LAKE;
    }
}

void Fish::InRnd() {
    for (size_t i = 0; i < MAX_NAME_LEN; ++i) {
        this->name[i] = char('a' + (this->rnd26.Get()-1));
    }
    this->name[MAX_NAME_LEN - 1] = '\0';
    this->weight = Animal::rnd26.Get();
    int type = Animal::rnd3.Get();
    if (type == 1) {
        p = Fish::RIVER;
    } else if (type == 2) {
        p = Fish::SEA;
    } else {
        p = Fish::LAKE;
    }
}

void Fish::Out(std::ofstream &ofst) {
    ofst << "This is Fish: name = "<<this->name<<", weight = "<<this->weight<<", place = ";
    if (this->p == 0){
        ofst<<"RIVER";
    } else if (this->p == 1){
        ofst<<"SEA";
    } else if (this->p == 2){
        ofst<<"LAKE";
    }
    ofst<<", codes / weight = " << this->CodesDividedByWeight()<<"\n";
}

void Fish::OutInFile(std::ofstream &ofst) {
    ofst<<"fish\n"<<this->name<<"\n"<<this->weight<<"\n"<<this->p<<"\n";
}

File: task_3/src/fish.py
from enum import Enum

from animal import *


class FishType(Enum):
    RIVER = "RIVER"
    SEA = "SEA"
    LAKE = "LAKE"


class Fish(Animal):

    def __init__(self):
        super().__init__()
        self.place = None

    def ReadFromFile(self, strArray, i):
        i = super().ReadFromFile(strArray, i)
        if (i >= len(strArray) or i == 0):
            return 0
        place = strArray[i]
        if place == FishType.RIVER.name:
            self.place = FishType.RIVER
        elif place == FishType.SEA.name:
            self.place = FishType.SEA
        elif place == FishType.LAKE.name:
            self.place = FishType.LAKE
        else:
            print("Wrong fish place")
            return 0
        i+=1
        return i

    def ReadRnd(self):
        super().ReadRnd()
        place = get_random_type()
        if place == 0:
            self.place = FishType.RIVER
        elif place == 1:
            self.place = FishType.SEA
        elif place == 2:
            self.place = FishType.LAKE

    def PrintToConsole(self):
        print("Fish: name: {}, weight: {}, place: {}, codes / weight: {}".format(self.name, self.weight, self.place.name, self.CodesDividedByWeight()))

    def Write(self, ostream):
        ostream.write("fish\n")
        ostream.write(str(self.name)+"\n")
        ostream.write(str(self.weight)+"\n")
        ostream.write(str(self.place.name)+"\n")

    def WriteToFile(self, ostream):
        ostream.write("Fish: name: {}, weight: {}, place: {}, codes / weight: {}\n".format(self.name, self.weight, self.place.name, self.CodesDividedByWeight()))

    def CodesDividedByWeight(self):
        return super().CodesDividedByWeight()
